doc/design/multiprocess.md

```diff
@@ -1,72 +1,200 @@
+
+## Example Use Cases and Flows
+
+### Retrieving a Block Hash
+
+Let’s walk through an example where the `bitcoin-wallet` process requests the hash of a block at a specific height from the `bitcoin-node` process. This example demonstrates the practical application of the IPC mechanism, specifically the interplay between C++ method calls and Cap’n Proto-generated RPC calls.
+
+1. **Initiation in bitcoin-wallet**
+   - The wallet process calls the `getBlockHash` method on a `Chain` object. This method is defined as a virtual function in [`src/interfaces/chain.h`](../../src/interfaces/chain.h).
+
+2. **Translation to Cap’n Proto RPC**
+   - The call to `getBlockHash` is translated into a Cap’n Proto RPC call. This translation is handled by code automatically generated by the `mpgen` tool, based on the [`chain.capnp`](../../src/ipc/capnp/chain.capnp)` file in [`src/ipc/capnp/`](../../src/ipc/capnp/).
+
+3. **Request Preparation and Dispatch**
+   - The generated `Chain` subclass in `bitcoin-wallet` populates a Cap’n Proto request with the `height` parameter.
+   - The request is sent to the `bitcoin-node` process.
+
+4. **Handling in bitcoin-node**
+   - Upon receiving the request, the `bitcoin-node` process reads the `height` value from the request.
+   - It then calls the `getBlockHash` method on its local `Chain` object with the provided `height`.
+
+5. **Response and Return**
+   - The `bitcoin-node` process takes the `getBlockHash` return value, encapsulates it in a Cap’n Proto response, and sends it back to the `bitcoin-wallet` process.
+   - The `bitcoin-wallet` process receives the response, extracts the block hash value, and returns it to the original caller.
+
+## Future Enhancements
+
+Further improvements are possible such as:
+
+- Separating indexes from `bitcoin-node`, and running indexing code in separate processes (see [indexes: Stop using node internal types #24230](https://github.com/bitcoin/bitcoin/pull/24230)).
+- Enabling wallet processes to listen for JSON-RPC requests on their own ports instead of needing the node process to listen and forward requests to them.
+- Automatically generating `.capnp` files from C++ interface definitions (see [Interface Definition Maintenance](#interface-definition-maintenance)).
+- Simplifying and stabilizing interfaces (see [Interface Stability](#interface-stability)).
+- Adding sandbox features, restricting subprocess access to resources and data (see [https://eklitzke.org/multiprocess-bitcoin](https://eklitzke.org/multiprocess-bitcoin)).
+- Using Cap'n Proto's support for [other languages](https://capnproto.org/otherlang.html), such as [Rust](https://github.com/capnproto/capnproto-rust), to allow code written in other languages to call Bitcoin Core C++ code, and vice versa (see [How to rustify libmultiprocess? #56](https://github.com/chaincodelabs/libmultiprocess/issues/56)).
+
+## Conclusion
```

Comment:
nit: This just repeats what is said in the Introduction part and isn't really needed for the purpose of this doc IMO